File: backend/Models/billModel.js
const mongoose = require("mongoose");

const billSchema = new mongoose.Schema({
  billId: { type: String, unique: true, required: true },
  table: { type: String, required: true },

  user: {
    fullName: {
      type: String,
      trim: true,
      default: "Walk-in / Guest",
    },
    mobile: {
      type: String,
      trim: true,
      default: "1234567890",
    },
  },

  orders: [
    {
      orderId: { type: String, required: true },
      items: [
        {
          name: { type: String, required: true },
          customization: { type: String, required: true },
          quantity: { type: Number, required: true },
          price: { type: Number, required: true },
        },
      ],
      total: { type: Number, required: true },
      status: { type: String, required: true },
      timestamp: { type: Date, required: true },
    },
  ],

  totalAmount: { type: Number, required: true },
  isGstApplied: { type: Boolean, required: true },

  paymentMethod: { type: String , required : false , default:"NONE"},
  transactionId: { type: String , required : false , default:"NONE"},

  status: { type: String, default: "Pending" },

  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now },
});



const Bill = mongoose.model("Bill", billSchema);

module.exports = Bill;

File: frontend/lib/models/bill_model.dart

import 'user_model.dart';
import 'order_model.dart';
class Bill {
  final String billId;
  final String table;
  // final UserModel user;
  final List<Order> orders;
  final double totalAmount;
  final bool isGstApplied;

  Bill({
    required this.billId,
    required this.table,
    // required this.user,
    required this.orders,
    required this.totalAmount,
    required this.isGstApplied,
  });

  Map<String, dynamic> toJson() => {
        'billId': billId,
        'table': table,
        // 'user': user.toJson(), // ✅ convert properly
        'orders': orders.map((e) => e.toJson()).toList(), // ✅ convert list
        'totalAmount': totalAmount,
        'isGstApplied': isGstApplied,
      };
}
